### PR TITLE
Waves

### PR DESCRIPTION
**Pull request by bahamut657**
@AnthonyMyatt  @fivdi, I've just completed and tested and documented the Wave section  of pigpio library.
New function calls available are:
gpioWaveChain 
gpioWaveAddSerial
gpioTick
gpioSetWatchdog

I will add 433D porting to example and create another repository implementing VirtualWire with pigpio

### CODE REVIEW

File: src/pigpio.cc [hunk 1]
```diff
@@ -763,9 +763,33 @@ NAN_METHOD(gpioWaveAddGeneric) {
     }
 }
     
-//NAN_METHOD(gpioWaveAddSerial) {
-//
-//}
+
+
+NAN_METHOD(gpioWaveAddSerial) {
+  if (info.Length() < 6 ||
```
Comment (AnthonyMyatt) [inline src/pigpio.cc]: Should be 7 (0-6). Could you please fix this.

File: src/pigpio.cc [hunk 2]
```diff
@@ -763,9 +763,33 @@ NAN_METHOD(gpioWaveAddGeneric) {
...
+       !info[0]->IsUint32() ||
+       !info[1]->IsUint32() ||
+       !info[2]->IsUint32() ||
+       !info[3]->IsUint32() ||
+       !info[4]->IsUint32() ||
+       !info[5]->IsUint32() ||
+       info[6]->IsNull()){
+      return Nan::ThrowError(Nan::ErrnoException(EINVAL, "gpioWaveAddSerial", ""));
+  }
+  unsigned user_gpio = info[0]->Uint32Value();
```
Comment (AnthonyMyatt) [inline src/pigpio.cc]: I know the original pigpio is inconsistent, but could you please stick to one style... preferably camel case.

File: pigpio.js [hunk 3]
```diff
@@ -376,10 +376,12 @@ Gpio.prototype.waveDelete = function(waveId) {
 	pigpio.gpioWaveDelete(waveId);
 	return this;
 };
-
 Gpio.prototype.waveTxSend = function(waveId, waveMode) {
 	return pigpio.gpioWaveTxSend(waveId, waveMode);
 };
+Gpio.prototype.waveAddSerial = function(user_gpio, baud, data_bits, stop_bits, offset, numBytes, str) {
+	return pigpio.gpioWaveAddSerial(user_gpio, baud, data_bits, stop_bits, offset, numBytes, str);
```
Comment (AnthonyMyatt) [inline pigpio.js]: I know the original pigpio is inconsistent, but could you please stick to one style... preferably camel case.

**Review by AnthonyMyatt — CHANGES_REQUESTED** [review]
Thank you for your contribution. Just a few minor changes needed.

File: src/pigpio.cc (hunk 1 above)
Comment (bahamut657) [reply src/pigpio.cc]: Fixed parameters check in 'gpioWaveAddSerial'

**Comment by bahamut657** [comment]
Syntax style corrected (camelCase  style). 
Fixed parameter check in gpioWaveAddSerial
@AnthonyMyatt, can we merge now to the Waves branch?

**Pull request merged**